### ISSUE TITLE
Planner Closed items should be hidden

### ISSUE THREAD

**Issue by joshuawilson** [openshiftio/openshift.io#4480]
##### Issue Overview

It is too easy to look at the planner list and think that a `closed` item is active. While planning, I almost never want to see finished work.

##### Expected Behaviour

When I look at the list of items, the closed ones should be hidden behind a filter of some kind. Github does this with a tab that toggles the filters between Open and Closed. 

##### Current Behaviour

Closed items are in the list

**Comment by joshuawilson** [comment]
Maybe that only works for the `Work Items` view. While in an iteration, I still see closed items.

**Comment by joshuawilson** [comment]
Nope, doesn't work there either. I don't know what it is supposed to do but it doesn't hide closed items.

**Comment by joshuawilson** [comment]
@serenamarie125 what do you think about a permanent filter or tab that has the `closed` issues?

**Comment by joshuawilson** [comment]
From @rootAvish 

> There is no point to having a 'resolved' status if both cannot be easily differentiated from each other in the overview by any means other than the "type" column. A basic workaround to it is to make them appear greyed out or put a "closed" icon or notification next to them.

**Comment by Veethika** [comment]
@joshuawilson I just confirmed, 'Show Completed' checkbox does hide closed items even in the iteration view. But if the tree view is enabled, it might still show the parent item that has been closed.
Also, there's a clear visual difference in the type weight for a closed item to differentiate it from regular items.
IMO we should be using one term at both places instead of - 'Completed' and 'Closed'.

**Comment by Veethika** [comment]
After discussing with Joshua it looks like after all there is a bug in the implementation.
@vikram-raj In the attached screenshot, the closed work items are appearing even if they don't have a child item that's in an active state.
@sunilmalagi there's no differentiation in the way the active and closed items are displayed. Did you provide any design for that earlier?
<img width="1675" alt="screenshot 2018-11-15 at 8 05 10 pm" src="https://user-images.githubusercontent.com/35127763/48560671-62bd7000-e914-11e8-9f47-4f89a6615fec.png">

**Comment by sunilmalagi** [comment]
@Veethika earlier we have limited no. of status and this was our proposal:
https://redhat.invisionapp.com/share/3HP6GZUBZSV#/290540725_States_Icons

Later there was need of more no. of status and we took the decision to takeout all the icons from the status, I also suggested to take out the generic icon which they are using right now (without consulting UX team), because it make no sense at all.

**Comment by vikram-raj** [comment]
>  In the attached screenshot, the closed work items are appearing even if they don't have a child item that's in an active state.

@Veethika  correct. Work item ```483``` and ```388``` are visible because its parent status is not closed and its sibling work item matches the applied filter.

Closed work item is not visible only if
- It is top level work item and none of its child work items match the applied filter.
- It is a child and none of there sibling matches the applied filter.

It is our current implemetation.
cc: @sahil143

**Comment by Veethika** [comment]
> @Veethika correct. Work item `483` and `388` are visible because its parent status is not closed and its sibling work item matches the applied filter.
> 
> Closed work item is not visible only if
> 
> * It is top level work item and none of its child work items match the applied filter.
> * It is a child and none of there sibling matches the applied filter.
@sahil143 @vikram-raj IMO if a work-item doesn't have a child that matches the filter, it's shouldn't appear in the result set at all. The only purpose of showing the closed WIs while the tree view enabled is to avoid breaking the tree structure. A sibling WIs status shouldn't be considered while surfacing closed items. Is there any other reasoning that I'm probably missing out on?

@sunilmalagi There's a difference in type treatment in the screenshot attached above. Some appear bold and others light. Are you aware of the intention behind that? Can we make the closed items appear different than the others?

**Comment by Veethika** [comment]
@sunilmalagi @joshuawilson  @serenamarie125 
From a visual design point of view, we need to answer the following questions:
1) Should we again have icons for each state? 
2) How to differentiate 'closed from not-closed WI', and 'WI with children from WI without children'?
3) Should there be a caret next to items without a child WI?

**Comment by nimishamukherjee** [comment]
Closed work items are displayed based on certain conditions. If a parent work item matches a criteria, all the children, closed and open ones, are displayed.
Why are we doing this - the backlog tree view serves two purposes:
1. Display matching results
2. Allow exploration

We need to change this to a feature request. Also, we can focus on the Closed Work Item behaviour under the Query tab.
